Make sure Include is in the main configuration file when drop-in directory is used

Jakuje opened the pull request:
This implements sanity check to make sure the configuration files on RHEL9 and Fedora (and now also Ubuntu 22.04), which are placed in drop-in directory are really applied by checking the include directive is present in the main configuration file.

There are also CI fixes, refactoring and the whole new support for new Ubuntu 22.04.

Jakuje commented:
@mattwillsher @richm can you have a look?

mattwillsher commented:
It looks ok to me.

We are getting quite a lot of logic and long conditionals now. I'm wondering how far we are off this needing a refactor or even rewrite

Jakuje commented:
> > We are getting quite a lot of logic and long conditionals now. I'm wondering how far we are off this needing a refactor or even rewrite
> 
> Yeah - probably time for a module

Thank you for the comments. You are right, the conditions are getting longer. I can think of some reshuffling specific tasks to different tasks files to make it shorter and more readable. Do you have some specific recommendation except for rewriting all of that into module?

richm commented:
> > > We are getting quite a lot of logic and long conditionals now. I'm wondering how far we are off this needing a refactor or even rewrite
> > 
> > 
> > Yeah - probably time for a module
> 
> Thank you for the comments. You are right, the conditions are getting longer. I can think of some reshuffling specific tasks to different tasks files to make it shorter and more readable. Do you have some specific recommendation except for rewriting all of that into module?

Not specifically, no

Jakuje commented:
> Not specifically, no

So far I only split the huge install.yml to smaller parts to get rid of some conditions. Not sure if it is much better. Any other suggestions how to make it more readable would be welcomed.

In the meantime, I added support for recent ubuntu 22.04, which also finally supports drop-in directory. I can split this work into separate PR if needed.

Jakuje commented:
@richm or @mattwillsher  is there something else to address or does it look ok to you?

mattwillsher commented:
Looks ok to me

richm commented on tasks/check_fips.yml:
```diff
@@ -0,0 +1,12 @@
+---
+- name: Check the kernel FIPS mode
+  slurp:
```
I've been told that there is no plan to require FQCN for built-ins in any foreseeable release of Ansible. I would suggest adding the suppression to .ansible-lint

Jakuje replied:
I already added it in previous PR (hopefully right). It makes the action not fail, but still reports these issues:

https://github.com/willshersystems/ansible-sshd/blob/master/.ansible-lint#L4

richm replied:
ack - lgtm

ep69 commented on README.md:
```diff
@@ -199,10 +204,12 @@ file that this role produces.
-and file owners are set to sane defaults. This is useful if the role is used
-in deployment stage to make sure the service is able to start on the first
-attempt. To disable this check, set this to empty list.
+are checked for presence and new keys are generated if they aremissing.
```
Typo: `aremissing`

Jakuje replied:
Thanks. Fixed and rebased.

Jakuje commented on README.md:
```diff
@@ -148,24 +145,32 @@ is *true*.
+When this path points to system drop-in directory (defined with internal
+variable `__sshd_drop_in_dir`), the main configuration file (defined with
+internal variable `__sshd_main_config_file`) is checked to contain the
+defailt `Include` directive from `__sshd_defaults` dict.
```
Thanks. Fixed and rebased.

Jakuje commented:
@mattwillsher do you want to take care of the release or should I draft something? I think it still involves the manual step of publishing the role on the galaxy.

mattwillsher commented:
To summarise for the release notes:

- adds support for Ubuntu 22.04
- improved support for RHEL9 and Fedora use of drop-in config
- genera quality improvements

Jakuje commented:
I would probably add points like this as there is a bit more changes since last release:

 * Support for Ubuntu 22.04 with drop-in directory (71eab11)
 * When using drop-in directory, verify the main configuration file has include directive (#178)
 * Tests improvements and general maintenance
 * Ensure the needed facts are available (e24ff31)
 * Filter out incompatible host keys on FIPS systems (#173)

https://github.com/willshersystems/ansible-sshd/compare/v0.14.1...master

mattwillsher commented:
v0.15.0 released and uploaded to Galaxy. Thanks all.
